src/sentry/tasks/store.py

```diff
@@ -286,6 +294,11 @@ def save_event(cache_key=None, data=None, start_time=None, event_id=None, **kwar
                 'description': exc.message,
             }
         )
+        quotas.refund(
+            Project.objects.get(id=project),
```

Comment:
Use `get_from_cache` here, and for `ProjectKey`.

We should also guard ourselves here against any `ObjectDoesNotExist` errors that will somehow be raised to make sure we don't break things. If that got raised here, it'd cause the whole task to fail, resulting in it retrying to save, causing bad things to happen since I don't think these would be fully idempotent.